File: SocketsJava/Services/IService.java
package agentie.services;

import agentie.model.Angajat;
import agentie.model.Excursie;
import agentie.model.Rezervare;

import java.util.Optional;

public interface IService {
    // Angajat
    void login(Angajat angajat, IObserver client) throws AgentieException;
    void logout(Angajat angajat) throws AgentieException;
    // Optional<Angajat> findAngajatByUsername(String username) throws AgentieException;

    // Excursie
    Iterable<Excursie> findAllExcursii() throws AgentieException;
    Optional<Excursie> updateExcursie(Excursie excursie) throws AgentieException;
    Iterable<Excursie> cautaExcursii(String numeObiectiv, Integer deLaOra, Integer panaLaOra) throws AgentieException;

    // Rezervare
      Optional<Rezervare> addRezervare(Rezervare rezervare) throws AgentieException;
}

File: SocketsCS/Services/IService.cs
#nullable enable
using System.Collections.Generic;
using Model;

namespace Services
{
    public interface IService
    {
        // Angajat
        void Login(Angajat angajat, IObserver client);
        void Logout(Angajat angajat);
        
        // Excursie
        IEnumerable<Excursie> FindAllExcursii();
        Excursie? UpdateExcursie(Excursie excursie);
        IEnumerable<Excursie> CautaExcursii(string numeObiectiv, int deLaOra, int panaLaOra);

        // Rezervare
        Rezervare? AddRezervare(Rezervare rezervare);
    }
}